### PR TITLE
Toggle button

### PR DESCRIPTION
**Pull request by adhiththiyan**
Current Problem
None

Screenshot 2024-08-24 123931

After Implementation
Functionalities added

Steps to Reproduce this issue
None

### CODE REVIEW

File: react/src/CallsToAction/ToggleButton/ToggleButton.js [hunk 1]
```diff
@@ -1,7 +1,16 @@
-import React from 'react'
+import React, { useState } from 'react'
 
-function ToggleButton() {
-  return <>ToggleButton goes here.</>
-}
+function ToggleButton({ onIcon, offIcon, overwriteClass}) {
+    const [toggle, setToggle] = useState(true)
```
Comment (send2alamelu) [inline react/src/CallsToAction/ToggleButton/ToggleButton.js:4]: by default its true, can this be prop driven?

```
 function ToggleButton({ toggled, onIcon, offIcon, overwriteClass}) {
 useState(toggled)
```

File: react/src/CallsToAction/ToggleButton/ToggleButton.js [hunk 2]
```diff
@@ -1,7 +1,16 @@
...
 
+    const buttonClasses = overwriteClass || 'px-2 py-2 rounded-md cursor-pointer';
+    const handleToggle =(e)=>{
+        setToggle(!toggle)
+    }
+  return(
+    <div className={`${buttonClasses}`} onClick={handleToggle}>
```
Comment (send2alamelu) [inline react/src/CallsToAction/ToggleButton/ToggleButton.js:11]: no need, can be like `className={buttonClasses}`

File: react/src/CallsToAction/ToggleButton/ToggleButton.js [hunk 3]
```diff
@@ -1,7 +1,16 @@
...
+import React, { useState } from 'react'
 
-function ToggleButton() {
-  return <>ToggleButton goes here.</>
-}
+function ToggleButton({ onIcon, offIcon, overwriteClass}) {
+    const [toggle, setToggle] = useState(true)
 
+    const buttonClasses = overwriteClass || 'px-2 py-2 rounded-md cursor-pointer';
+    const handleToggle =(e)=>{
```
Comment (send2alamelu) [inline react/src/CallsToAction/ToggleButton/ToggleButton.js:7]: format this

File: react/src/CallsToAction/ToggleButton/ToggleButton.js [hunk 4]
```diff
@@ -1,7 +1,16 @@
...
+        setToggle(!toggle)
+    }
+  return(
+    <div className={`${buttonClasses}`} onClick={handleToggle}>
+        {toggle ? onIcon: offIcon}
```
Comment (send2alamelu) [inline react/src/CallsToAction/ToggleButton/ToggleButton.js:12]: what happens if onIcon / offIcon is not exist

**Review by send2alamelu — CHANGES_REQUESTED** [review]

**Comment by adhiththiyan** [comment]
Completed the changes kindly review it

**Pull request merged**